spec/contact_spec.rb

```diff
@@ -2,19 +2,121 @@
 
 describe Amorail::AmoContact do
   before { mock_api }
-  
+
+  let(:contact) { described_class.new(name: "test") }
+
   describe "validations" do
-    it { should validate_presence_of(:name)}
+    it { should validate_presence_of(:name) }
   end
 
-  describe "#save" do
-    let(:contact) { Amorail::AmoContact.new(name: "test") }
+  describe ".attributes" do
+    subject { described_class.attributes }
+
+    it_behaves_like 'entity_class'
+
+    specify { is_expected.to include(:name, :company_name) }
+  end
+
+  describe ".properties" do
+    subject { described_class.properties }
+
+    specify { is_expected.to include(:email, :phone, :position) }
+  end
+
+  describe "#params" do
+    let(:contact) do
+      described_class.new(
+        name: 'Tester',
+        company_name: 'Test inc',
+        phone: '12345678',
+        email: 'test@mala.ru',
+        position: 'CEO'
+      )
+    end
+
+    subject { contact.params }
+
+    specify { is_expected.to include(:last_modified) }
+    specify { is_expected.to include(name: 'Tester') }
+    specify { is_expected.to include(company_name: 'Test inc') }
+
+    it "contains email property" do
+      prop = subject[:custom_fields].detect { |p| p[:id] == "1460591" }
+      expect(prop).not_to be_nil
+      expect(prop[:values].first[:value]).to eq 'test@mala.ru'
+      expect(prop[:values].first[:enum])
+        .to eq described_class.properties[:email][:enum]
+    end
+
+    it "contains phone property" do
+      prop = subject[:custom_fields].detect { |p| p[:id] == "1460589" }
+      expect(prop).not_to be_nil
+      expect(prop[:values].first[:value]).to eq '12345678'
+      expect(prop[:values].first[:enum])
+        .to eq described_class.properties[:phone][:enum]
```

Comment:
Style/DotPosition: Place the . on the previous line, together with the method call receiver.